src/Raven.Server/Rachis/ClusterTransactionWaiter.cs

```diff
@@ -2,29 +2,51 @@
 using System.Collections.Concurrent;
 using System.Threading;
 using System.Threading.Tasks;
+using Raven.Server.Documents;
+using Raven.Server.Extensions;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Server.Rachis
 {
-    public sealed class ClusterTransactionWaiter : AsyncWaiter<ClusterTransactionCompletionResult>
+    public sealed class ClusterTransactionWaiter : AsyncWaiter<long?>
```

Comment:
let's keep the `ClusterTransactionCompletionResult` class here, but change the internals of it

Reply:
or probably `ClusterTransactionResult` is even better?

Reply:
let's replace all places of `long?` with `ClusterTransactionResult`